src/machines/activity_state_machine.rs

```diff
@@ -735,6 +723,16 @@ fn new_timeout_failure(dat: &SharedState, attrs: ActivityTaskTimedOutEventAttrib
     }
 }
 
+fn convert_payloads(
```

Comment:
This is general enough that it doesn't belong here. Probably in proto or protosext module

Reply:
It is, the only thing that concerns me is WFMachinesError, which is specific to the state machine code. Should we generalize it too, or what do you think?

Reply:
The function could just go in the machines module